arlas-core/src/main/java/io/arlas/server/utils/GeoUtil.java

```diff
@@ -167,23 +168,29 @@ public static void checkWKT(String wktString) throws InvalidParameterException {
     }
 
     /**
-     *
-     * @param polygon CW oriented geometry
-     * @return List of geometries with longitudes between -180 and 180 and a join of thoses geometries in a MultiPolygon
+     * Splits the CW oriented polygon into a list of polygons with longitudes between -180 and 180 and with a longitude extent <= 180
+     * @param polygon CW oriented polygon
+     * @return The list of polygons and a join of these polygons in a MultiPolygon as a Tuple
      * @throws ArlasException
      */
-    public static Tuple2<List<Polygon>, Geometry> splitGeometryOnDateline(Polygon polygon) throws ArlasException {
+    public static Tuple2<List<Polygon>, Geometry> splitPolygon(Polygon polygon) throws ArlasException {
         Envelope envelope = polygon.getEnvelopeInternal();
         List<Polygon> geometries = new ArrayList<>();
         double envelopeEast = envelope.getMaxX();
         double envelopeWest = envelope.getMinX();
+        GeometryFactory geometryFactory = new GeometryFactory(new PrecisionModel(), 4326);
+        Geometry middleWest = geometryFactory.toGeometry(new Envelope(-180, 0, -90, 90));
+        Geometry middleEast = geometryFactory.toGeometry(new Envelope(0, 180, -90, 90));
         if (envelopeEast <= 180 && envelopeWest >= -180) {
             /** longitudes between -180 and 180**/
-            geometries.add(polygon);
+            if ((envelopeEast - envelopeWest) > 180) {
+                geometries = new ArrayList<>(Arrays.asList(middleWest.intersection(polygon), middleEast.intersection(polygon))
```

Comment:
Do you really need the `new ArrayList<>(...)`? (same comment for other envelopeEast / envelopeWest cases)

Reply:
You're right, no need in this case

Reply:
done